### DISCUSSION TITLE
Video saving using imagewriter and ffmpeg

### DISCUSSION THREAD

**Discussion by Lindsey-Christinger** [bonsai-rx/bonsai#1843] [Q&A]
Hi! I am new to Bonsai and am having issues saving the videos. I can see the camera stream while running, and ImageWriter can also be visualized. It then uses the ffmpeg to run the images through a pipe instance and uses ffmpeg. The parameters we are using for FFMPEG are: -y -f rawvideo -vcodec rawvideo -s 640x480 -r 25 -pix_fmt gray -i \\.\pipe\video -c:v libx264 -b:v 2M -maxrate 2M -bufsize 1M C:\Users\User\Desktop\Lindsey\bonsai_test\bonsai_video\bonsai . I would appreciate any help with this. The other functions such as saving of timestamp information are working. 
![image](https://github.com/bonsai-rx/bonsai/assets/112510219/9ad4abe4-f313-472f-8d6c-fdf4c2f12e84)
![image](https://github.com/bonsai-rx/bonsai/assets/112510219/231f3237-06e6-4185-9eef-e930091a5c8f)
![image](https://github.com/bonsai-rx/bonsai/assets/112510219/496a9460-52d0-4261-b8a6-65191cdabeef)
![image](https
[For_github.zip](https://github.com/user-attachments/files/15809125/For_github.zip)
://github.com/bonsai-rx/bonsai/assets/112510219/22716ea1-a7bf-4c40-bbca-af3483d005d3)
In the Zip folder I have added the workflow. I know this workflow works on another PC in the lab but not the PC I want to use for my experiments. Thanks!

**Comment by bruno-f-cruz** [comment]
I didnt get what the question/problem is, can you explain further? thanks!

**Reply by Lindsey-Christinger** [reply]
Yes, I am trying to record a video using ImageWriter and FFMPEG with these parameters:  -y -f rawvideo -vcodec rawvideo -s 640x480 -r 25 -pix_fmt gray -i \.\pipe\video -c:v libx264 -b:v 2M -maxrate 2M -bufsize 1M C:\Users\User\Desktop\Lindsey\bonsai_test\bonsai_video\bonsai. ImageWriter is directed toward  \\.\pipe\video. The camera stream is visible while I run bonsai and information about timestamps are correctly being recorded for each video but the actual video itself is not saving. This workflow also works on another PC in the lab to record both video and timestamp information. I don't know what the issue might be and would appreciate any direction! Let me know if I can clarify the problem more.

**Reply by bruno-f-cruz** [reply]
Do you see anything being printed out to the console running python?

**Reply by Lindsey-Christinger** [reply]
I am still new to bonsai, how can I check if anything is being printed?

**Reply by Lindsey-Christinger** [reply]
I don't see anything printed. Sometimes when I run bonsai I get the message: The procedure entry point vkBindImageMemory2 could not be located in the dynamic link library vulkan-1.dll.
